multiqc/modules/bcl2fastq/bcl2fastq.py

```diff
@@ -166,14 +166,13 @@ def parse_file_as_json(self, myfile):
             rlane = run_data[lane]
 
             # Add undetermined barcodes
-            try:
-                unknown_barcode = content['UnknownBarcodes'][l - 1]['Barcodes']
-            except IndexError:
-                unknown_barcode = next(
-                    (item['Barcodes'] for item in content['UnknownBarcodes'] if item['Lane'] == 8),
-                    None
-                )
-            run_data[lane]['unknown_barcodes'] = unknown_barcode
+            for lane_data in content.get("UnknownBarcodes", list()):
+                if lane_data["Lane"] == l:
+                    unknown_barcode = lane_data["Barcodes"]
+                    break
+            else:
+                unknown_barcode = dict()
```

Comment:
Instead of `break` / `else` syntax, I'd prefer to just initialise this empty `dict()` before the `for` loop and then it can be overwritten. The `break` / `else` syntax isn't super common and I'm worried that future edits might lead to a bug that could be quite tricky to track down.

Reply:
Makes sense. Done.